pg_shard.c

```diff
@@ -92,9 +95,11 @@ static Oid ExtractFirstDistributedTableId(Query *query);
 static bool ExtractRangeTableEntryWalker(Node *node, List **rangeTableList);
 static List * DistributedQueryShardList(Query *query);
 static bool SelectFromMultipleShards(Query *query, List *queryShardList);
+static Query * BuildLocalQuery(Query *query, List *localRestrictList);
```

Comment:
I indeed wanted to change the order but knew doing so would dirty the diff (it would just highlight the whole method in green). You could always view each commit separately, but that makes commenting harder. I'm going to clean them up in exactly this way once we hash out the implementation issues first.